src/services/request-presentation-service.ts

```diff
@@ -1,99 +1,135 @@
 import type { Request } from '../models/request.js'
 import { InvalidRequestStatusError, RequestStatus } from '../models/request.js'
 import type { IAnchorRepository } from '../repositories/anchor-repository.type.js'
+import { AnchorWithRequest } from '../repositories/anchor-repository.type.js'
+import type { IMerkleCarService } from './merkle-car-service.js'
+import type { WitnessService } from './witness-service.js'
+import type { CAR } from 'cartonne'
+import {
+  CASResponse,
+  CompleteCASResponse,
+  NotCompleteCASResponse,
+  CommitPresentation,
+} from '../ancillary/anchor-codecs.js'
+import { RequestStatusName } from '@ceramicnetwork/anchor-utils'
+import { CID } from 'multiformats/cid'
+import { StreamID } from '@ceramicnetwork/streamid'
+import type { OutputOf } from 'codeco'
+import { LRUCache } from 'lru-cache'
 
-export type CommitPresentation = {
-  content: { path: string | undefined; prev: string; proof: string | undefined }
-  cid: string
-}
-
-export type NotCompletedRequestPresentation = {
-  createdAt: number
-  streamId: string
-  id: string
-  message: string
-  status: string
-  cid: string
-  updatedAt: number
-}
+const NAME_FROM_STATUS = {
+  [RequestStatus.REPLACED]: RequestStatusName.REPLACED,
+  [RequestStatus.FAILED]: RequestStatusName.FAILED,
+  [RequestStatus.PENDING]: RequestStatusName.PENDING,
+  [RequestStatus.PROCESSING]: RequestStatusName.PROCESSING,
+  [RequestStatus.READY]: RequestStatusName.READY,
+  [RequestStatus.COMPLETED]: RequestStatusName.READY,
+} as const
 
-export type CompletedRequestPresentation = NotCompletedRequestPresentation & {
-  anchorCommit: CommitPresentation
-}
-
-export type RequestPresentation = NotCompletedRequestPresentation | CompletedRequestPresentation
+const WITNESS_CAR_CACHE = 1000 // ~2MiB if one witness car is 2KiB
 
 /**
  * Render anchoring Request as JSON for a client to consume.
  */
 export class RequestPresentationService {
-  static inject = ['anchorRepository'] as const
+  static inject = ['anchorRepository', 'merkleCarService', 'witnessService'] as const
 
-  constructor(private readonly anchorRepository: IAnchorRepository) {}
+  private readonly cache = new LRUCache<string, CAR>({ max: WITNESS_CAR_CACHE })
 
-  /**
-   * Rich JSON of a request.
-   *
-   * @param request - Request to be rendered as JSON.
-   */
-  async body(request: Request): Promise<RequestPresentation> {
+  constructor(
+    private readonly anchorRepository: IAnchorRepository,
+    private readonly merkleCarService: IMerkleCarService,
+    private readonly witnessService: WitnessService
+  ) {}
+
+  async witnessCAR(anchor: AnchorWithRequest | null): Promise<CAR | null> {
+    if (!anchor) return null // TODO Add metric
+    const cacheKey = anchor.cid.toString()
+    const fromCache = this.cache.get(cacheKey)
+    if (fromCache) return fromCache
+    const merkleCAR = await this.merkleCarService.retrieveCarFile(anchor.proofCid)
+    if (!merkleCAR) return null // TODO Add metric
```

Comment:
we should probably log a warning for this case as it isn't expected

Reply:
```ts
logger.warn(`No Merkle CAR found for anchor ${anchor.cid}`)
```